src/fields/fq2.rs

```diff
@@ -255,15 +256,14 @@ impl FieldElement for Fq2 {
         // "High-Speed Software Implementation of the Optimal Ate Pairing
         // over Barreto–Naehrig Curves"; Algorithm 8
 
-        (self
-            .c0
-            .cpu_mul(self.c0)
-            .cpu_sub((self.c1.cpu_mul(self.c1)).cpu_mul(fq_non_residue())))
-        .inverse()
-        .map(|t| Fq2 {
-            c0: self.c0.cpu_mul(t),
-            c1: (self.c1.cpu_mul(t)).cpu_neg(),
-        })
+        self.c0
+            .mul(self.c0)
+            .sub((self.c1.mul(self.c1)).mul(fq_non_residue()))
+            .inverse_unconstrained()
+            .map(|t| Fq2 {
+                c0: self.c0.cpu_mul(t),
+                c1: (self.c1.cpu_mul(t)).cpu_neg(),
+            })
```

Comment:
And why are we using `cpu_mul` here?

It'd be great to include a comment for why we're using different functions from the orig

Reply:
u right these can be mul and neg

basically there are redundant checks for which os we're in. 

In general, I think differentiating between inverse and inverse unconstrained is a bad idea. The FieldElement trait should just expose inverse, and it should compile to the specialized unconstrained implementation or the default implementation depending on the target os. Simplifying the interface in this way will make errors like this impossible, and will probably reduce cycle count too.